### DISCUSSION TITLE
Enabling display of Slideover when clicking on a row within the  public Table Builder

### DISCUSSION THREAD

**Discussion by snarexx** [filamentphp/filament#10562] [Help] [answered]
How can I enable the display of a SlideOver when clicking on a row within the Table Builder in a public table.  I have already implemented clickable rows which redirect to a URL rather than displaying as slideover. I also have a "View" button. When I click that button in the same table, it displays the slideover.

```
use Filament\Tables\Table;
use Illuminate\Database\Eloquent\Model;
 
public function table(Table $table): Table
{
    return $table
        ->recordUrl(
            fn (Model $record): string => route('race', ['record' => $record]),
        );
} 
```

I have this action, it works for "View" button. But when I click the row, it opens in a new page, not slideover. I guess I should add it somewhere over here
```

          ->actions([
x                Tables\Actions\ViewAction::make()->slideOver(),
            ])
```

**Comment by xerk** [comment]
I think this i might be helpful 

```
->actions([
     Tables\Actions\EditAction::make()
          ->slideOver(),
])
```

https://v2.filamentphp.com/tricks/use-a-slide-over-instead-of-a-modal

**Reply by snarexx** [reply]
I'm already doing this, I don't want it to redirect to edit page, I want it to show the ViewAction::make() as I do with the button. @xerk

**Comment by mahmudz** [comment]
Is this what you wanted?

https://github.com/filamentphp/filament/assets/22455598/3d6bacb4-52ed-4f49-9351-90a915131a79

**Reply by snarexx** [reply]
Exactly But this is table builder inside a panel builder, not table builder open to public. My page is a public page. 
My panel builder works like this but table builder that open to public is not
In public view button display slideover but when I click row it doesn't.

**Reply by snarexx** [reply]
@mahmudz

**Reply by mahmudz** [reply]
in actions array
```php
->actions([
    Tables\Actions\ViewAction::make()
        ->infolist([
            ...ViewJob::getEntries()
        ])
        ->slideOver(),
])
```

Set table record url to null
```php
->recordUrl(null)
```

Also make sure the view route doesnt exists in `getPages` method
Let me know if it works.

> You did send me upwork invitation though, but i declined it due to very low price.

**Reply by snarexx** [reply]
@mahmudz for public "Table Builder" getPages doesn't exists and this is for panel builder. This part is the main problem here.

**Reply by mahmudz** [reply, marked as answer]
@snarexx then just chain `->recordAction(ViewAction::class)` like this.

**Reply by snarexx** [reply]
@mahmudz  it just worked. Thanks so much

**Reply by mahmudz** [reply]
@snarexx I think you marked the wrong comment as answer. It should be this https://github.com/filamentphp/filament/discussions/10562#discussioncomment-7975955
